lib/matplotlib/patches.py

```diff
@@ -1037,13 +1037,24 @@ def _update_path(self):
                              f"`len(values) = {self._values.size}` and "
                              f"`len(edges) = {self._edges.size}`.")
         verts, codes = [], []
-        for idx0, idx1 in cbook.contiguous_regions(~np.isnan(self._values)):
+
+        _nan_mask = np.isnan(self._values)
+        if self._baseline is not None:
+            _nan_mask |= np.isnan(self._baseline)
+        for idx0, idx1 in cbook.contiguous_regions(~_nan_mask):
             x = np.repeat(self._edges[idx0:idx1+1], 2)
             y = np.repeat(self._values[idx0:idx1], 2)
-            if self.baseline is not None:
-                y = np.hstack((self.baseline, y, self.baseline))
-            else:
+            if self._baseline is None:
                 y = np.hstack((y[0], y, y[-1]))
+            elif self._baseline.ndim == 0:  # single baseline value
```

Comment:
is this actually better than just making the baseline always be a vector?

Reply:
It's a bit of different behaviour. Passing a vector will close the unfilled shape, whereas this will not.

Reply:
OK, thats not clear from the docs, I don't think.  probably need something like, passing baseline as a vector will add a bottom line to the patch?

Reply:
I've added that in the last commit https://github.com/matplotlib/matplotlib/pull/18511/files#diff-7e79bc5b4cdd21de353697e9ada248b7R6735-R6736

Reply:
OK, I think that should somehow be in the baseline part of the docstring?  I definitely missed it up there....